src/transformers/trainer.py

```diff
@@ -3603,6 +3603,9 @@ def training_step(
                 scaled_loss.backward()
         else:
             loss *= self.args.gradient_accumulation_steps
+            if (self.args.average_tokens_across_devices and num_items_in_batch is not None and
+                    self.args.world_size > 1):
+                loss *= self.args.world_size
```

Comment:
Both of these chunks I think can be under a `if num_items_in_batch is not None`, since both need to be valid for the `loss *= self.args.gradient_accumulation_steps`

Reply:
I think gradient accumulation is orthogonal to DDP, and used a new if statement. Please check my code. Thanks.

Reply:
It is, it's a matter of `self.model.accepts_loss_kwargs`